dev/components/stock-popup.js

```diff
@@ -0,0 +1,217 @@
+          <h3 class="text-headline popup-stock-name"></h3>
+          <table>
+            <tbody>
+            </tbody>
+          </table>
+          <div class="popup-chart-container">
+            <div class="icon-loading">
+              <i class="fa fa-spinner fa-pulse fa-3x fa-fw"></i>
+            </div>
+            <canvas id="popup-chart" width="700" height="320"></canvas>
+          </div>
+          <button class="button btn-popup-watchlist">Add to watchlist</button>
+        </div>
+      </div>
+    `;
+    this.$mainContainer.prepend(popupModal);
+  }
+
+
+  // RENDER STOCK CONTENT FOR POPUP
+  renderPopUpContent(companyId, companyName) {
+
+    // check if there's locally stored data before making Ajax request
+    if (store.get(`${companyId}`)) {
+      this.renderStockInfo(companyId, companyName);
+      this.renderGraph(companyId);
+    }
+    else {
+      this.getPrice(companyId, companyName);
+    }
+
+    // disable closing of viewer upon click on popup container
+    this.$popupContentContainer.on('click', function(event) {
+      event.stopPropagation();
+    });
+
+    // remove popup modal
+    this.$popupContainer.on('click', function() {
+      $(this).remove();
+    });
+  }
+
+
+  // GET COMPANY STOCK DATA
+  getPrice(companyId, companyName) {
+    // display loading icon
+    this.$loadingIcon.addClass('is-visible');
+    $.ajax({
+      url: `https://www.quandl.com/api/v3/datasets/WIKI/${companyId}/data.json?api_key=tskzGKweRxWgnbX2pafZ`,
+      dataType: 'json',
+      success: (data) => {
+        // store company data
+        store.set(`${companyId}`, data);
+
+        // render stock info
+        this.renderStockInfo(companyId, companyName);
+
+        // render graph
+        this.renderGraph(companyId);
+      },
+      complete: () => {
+        this.$loadingIcon.removeClass('is-visible');
+      }
+    });
+  }
+
+
+  // RENDER TABLE WITH STOCK INFO
+  renderStockInfo(companyId, companyName) {
+    const stockData = store.get(`${companyId}`);
+
+    // get stock info from local storage
+    let closePrice = stockData.dataset_data.data[0][4];
+    let openPrice = stockData.dataset_data.data[0][1];
+    let low = stockData.dataset_data.data[0][3];
+    let high = stockData.dataset_data.data[0][2];
+    let volume = stockData.dataset_data.data[0][5];
+
+    // render stock name
+    this.$stockName.text(companyName);
+
+    let row = `
+      <tr>
+        <td class="key">Close</td>
+        <td class="val">${closePrice}</td>
+      </tr>
+      <tr>
+        <td class="key">Open</td>
+        <td class="val">${openPrice}</td>
+      </tr>
+      <tr>
+        <td class="key">Day's Range</td>
+        <td class="val">${low} - ${high}</td>
+      </tr>
+      <tr>
+        <td class="key">Volume</td>
+        <td class="val">${volume}</td>
+      </tr>
+    `;
+    this.$tbody.append(row);
+
+    // add stock id to watchlist button
+    this.$watchlistButton.attr('id', `${companyId}`);
```

Comment:
I originally did this so that I could take the button id onclick and use it to add/remove the stock from the watchlist. But the code is gone now since I registered the companyId in the constructor and can grab the id from there. Rookie mistake! haha